src/Packaging/Packaging.csproj

```diff
@@ -134,6 +134,102 @@
+        
+        // See http://petlibrary.tripod.com/ZIP.HTM
+          byte[] expectedHeader = new byte[] { 0x50, 0x4B, 0x03, 0x04, 0x14, 0x00, 0x00, 0x08 };
+
+          var data = System.IO.File.ReadAllBytes(FullFilePath);
+          for(int i = 0; i < expectedHeader.Length; i++)
+          {
+              if (data[i] != expectedHeader[i])
+              {
+                  Log.LogError("Invalid zip artifact: {0}. Unexpected value in zip header at index {1}",
+                    FullFilePath, i);
+                  return false;
+              }
+          }
+
+        return true;
+        ]]>
+      </Code>
+    </Task>
+  </UsingTask>
+  
+  <!-- There are MS-supplied Unzip and ZipDirectory tasks.
+       See https://docs.microsoft.com/en-us/visualstudio/msbuild/unzip-task?view=vs-2017
+       
+      However, the Zip task automatically changes forward-slashes to backslashes, which causes failures
+      when the final zipped artifacts are used on non-Windows machines.
+      So, we use a custom inline task that preserves any forward-slashes.
+      Note: the built-in ZipDirectory task uses a compression level of "optimal". We're currently using "fastest".
+-->
+  <UsingTask TaskName="CustomUnzip" TaskFactory="CodeTaskFactory" AssemblyFile="$(MSBuildToolsPath)\Microsoft.Build.Tasks.v12.0.dll">
+    <ParameterGroup>
+      <SourcePath ParameterType="System.String" Required="true" />
+      <SourceFileName ParameterType="System.String" Required="true" />
+      <TargetPath ParameterType="System.String" Required="true" />
+    </ParameterGroup>
+    <Task>
+      <Reference Include="System.IO.Compression.FileSystem" />
+      <Code Type="Fragment" Language="cs">
+        <![CDATA[    
+    var fullSourcePath = System.IO.Path.Combine(SourcePath, SourceFileName);
+    System.IO.Compression.ZipFile.ExtractToDirectory(fullSourcePath, TargetPath);
+        ]]>
+      </Code>
+    </Task>
+  </UsingTask>
+
+  <UsingTask TaskName="CustomZipDirectory" TaskFactory="CodeTaskFactory" AssemblyFile="$(MSBuildToolsPath)\Microsoft.Build.Tasks.v12.0.dll">
+    <ParameterGroup>
+      <SourceDirectoryPath ParameterType="System.String" Required="true" />
+      <TargetFilePath ParameterType="System.String" Required="true" />
+    </ParameterGroup>
+    <Task>
+      <Reference Include="System.IO.Compression.FileSystem" />
+      <Code Type="Class" Language="cs">
+        <![CDATA[    
+    public class CustomZipDirectory : Microsoft.Build.Utilities.Task
+    {
+        [Microsoft.Build.Framework.Required]
+        public string SourceDirectoryPath { get; set; }
+
+        [Microsoft.Build.Framework.Required]
+        public string TargetFilePath { get; set; }
+
+        private class MyEncoder : System.Text.UTF8Encoding
+        {
+            public MyEncoder() : base(true) { }
+            public override byte[] GetBytes(string s)
+            {
+              // Replace backslashes with forward-slashes to make sure the zip
```

Comment:
The following method is using a 4 spaces indentation and here 2, could you make sure to use only 1 style please.